doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,130 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
 are given in Chapter :ref:`extensionsofgallina`. How to customize the syntax
 is described in Chapter :ref:`syntaxextensionsandinterpretationscopes`.
 
+.. insertgram term fixannot
+
 .. productionlist:: coq
-   term             : forall `binders` , `term`
-                    : fun `binders` => `term`
-                    : fix `fix_bodies`
-                    : cofix `cofix_bodies`
-                    : let `ident` [`binders`] [: `term`] := `term` in `term`
-                    : let fix `fix_body` in `term`
-                    : let cofix `cofix_body` in `term`
-                    : let ( [`name` , … , `name`] ) [`dep_ret_type`] := `term` in `term`
-                    : let ' `pattern` [in `term`] := `term` [`return_type`] in `term`
-                    : if `term` [`dep_ret_type`] then `term` else `term`
-                    : `term` : `term`
-                    : `term` <: `term`
-                    : `term` :>
-                    : `term` -> `term`
-                    : `term` `arg` … `arg`
-                    : @ `qualid` [`term` … `term`]
-                    : `term` % `ident`
-                    : match `match_item` , … , `match_item` [`return_type`] with
-                    :   [[|] `equation` | … | `equation`] end
-                    : `qualid`
-                    : `sort`
-                    : `num`
-                    : _
-                    : ( `term` )
-   arg              : `term`
-                    : ( `ident` := `term` )
-   binders          : `binder` … `binder`
-   binder           : `name`
-                    : ( `name` … `name` : `term` )
-                    : ( `name` [: `term`] := `term` )
-                    : ' `pattern`
-   name             : `ident` | _
-   qualid           : `ident` | `qualid` `field`
-   sort             : SProp | Prop | Set | Type
-   fix_bodies       : `fix_body`
-                    : `fix_body` with `fix_body` with … with `fix_body` for `ident`
-   cofix_bodies     : `cofix_body`
-                    : `cofix_body` with `cofix_body` with … with `cofix_body` for `ident`
-   fix_body         : `ident` `binders` [`annotation`] [: `term`] := `term`
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+   term : `binder_constr`
```

Comment:
`binder_constr` should be expanded here, instead of being defined as a separate token.

Reply:
Really?  We already have 120 lines of grammar here.  `binder_constr` is 60 lines.  For readability, I think we want the chunks even smaller if possible.  There is some tradeoff here.  If you're reading the HTML output, you can easily click on the symbol to get the definition.  A user-friendly introduction could briefly mention the available constructs.

`binder_constr` seems a reasonable term for the `fun`, `forall` and `let` constructs, but the ones I show below don't seem to fit the concept of "binder".  I would go for eventually moving them into a separate nonterminal and a different sub-subsection of the chapter:

```
                 : if `term` `return_type_opt` then `term` else `term`
                 : `single_fix`
                 : `single_fix` with `fix_body_list` for `ident`
                 : if `term` is `ssr_dthen` `ssr_else`      (ssr plugin)
                 : if `term` isn't `ssr_dthen` `ssr_else`      (ssr plugin)
```

> instead of being defined as a separate token

nonterminal, actually.  I make the same mistake at times.

Reply:
> `binder_constr` seems a reasonable term for the `fun`, `forall` and `let` constructs

But these are very basic things that define what a term is, so it really does not make sense to cut this further away.